docs/design/solutions/path-validation/path-validation.md

```diff
@@ -0,0 +1,80 @@
+# Path Validation
+
+## Motivation
+The main use case is to allow users to validate if some arbitrary flow path is possible to create with the given 
+constraints without actually creating this flow. A path here is a sequence of nodes, that is a sequence of switches
+with in and out ports. Constraints are various parameters such as max latency, bandwidth, encapsulation type, 
+path computation strategy, and other.
+
+## Implementation proposal
+Reuse PCE components (specifically extend AvailableNetwork) for the path validation in the same manner as it is used for
+the path computation. Use nb worker topology, PathService, and PathsBolt because the path validation is read-only.
+
+_An alternative approach is to use FlowValidationHubBolt and the resource manager. This might help to mitigate concurrency 
+issues related to changes in repositories during a path validation process._
+
+There is no locking of resources for this path and, therefore, no guarantee that it will be possible to create this flow 
+after validation in the future.
+
+## Northbound API
+
+REST URL: ```/v2/network/validate-path```, method: ```GET```
+
+A user is required to provide a path represented by a list of nodes. Nodes must be ordered from start to end, 
+each next element is the next hop. A user can add optional parameters.
+
+### Request Body
+```json
+{
+  "encapsulation_type": "TRANSIT_VLAN|VXLAN",
+  "max_latency": 0,
+  "max_latency_tier2": 0,
+  "path_computation_strategy": "COST|LATENCY|MAX_LATENCY|COST_AND_AVAILABLE_BANDWIDTH",
```

Comment:
I would suggest to add couple fields here:
1. `max_bandwidth`
2. `reuse_flow_resources: {flow_id}` (you can change this name)
Last one is needed for following case:
Topology:
```
A-------B-------C
        |
        D 
```
We have existing flow `flow1` with path ABC
and we want to check if this flow can be on path ABD, but ISL AB was already occupied by flow1. So it wouldn't enough bandwidth on it to check path ABD. So if user specified `reuse_flow_resources` the API must ignore resources of this flow. (like we do in case of flow Update. TO find new path we ignore current path of a flow)